### PR TITLE
Blood Cult DLC 1: Make It an Actually Playable Game (Mode)

### PR DESCRIPTION
**Pull request by Remuchi**
# Description

You don't actually have to pay for it, y'know?
---

# TODO

- [x] Fix bugs from discord thread. 

---

<details><summary><h1>Media</h1></summary>
<p>

Example Media Embed

</p>
</details>

---

# Changelog

:cl:
- add: In-game guide book to kickstart your sinister activities.
- add: Constructs now have abilities.
- add: Rending rune and apocalypse rune now should only be placed in the specific spots on maps. Needs to be mapped.
- add: Veil Shifter now displays how much charges it has when examining.
- add: Cult runes now have descriptions. Also stating how much invokers required for each rune.
- add: Blood rites can now be dropped&deleted.
- add: Blood rites now suck... blood in 0.5 tiles radius.
- remove: Non-cultists can no longer examine runes.
- fix: Fixed Cult Objective Target selection. You can (and should) sacrifice your own people now.
- fix: Non cultists can no longer use veil shifter.
- fix: Teleport spell is no more a cheap rip-off and now actually teleports.
- fix: Timed Factories can't no more produce infinite number of entities.
- fix: Offering rune should now properly convert someone.
- fix: Sacrificing body with mind now properly transfers their mind to soul shard.
- fix: Shadow Shackles now cuffs the target instead of the caster (lmao).

### CODE REVIEW

**Comment by BlueHNT** [comment]
Q1: Does offering rune only turn people into gems? Can you not convert?
Q2: Is the offering rune supposed to be the conversion one even?

N1: Shackles just straight up don't work. I hope this fixes them.
N2: Spear doesn't get recalled if stuck in wall, can be picked up by others and thrown back, ow.

**Comment by angelofallars** [comment]
Bugs reported from the last Deepstation round (some of these might have been fixed already):

- Conversion not working, and only turning people into soul shards works
- Cult constructs and soul shards don't have languages set up for them, so they speak only speak Universal and can't understand Eldritch

**Comment by rbertoche** [comment]
This is starting to grow conflicts. I am going to make a merge with the conflict and if necessary open a new PR to merge this as is.

We've been running this on Pirata for a while now and although we haven't been able to test bloodcult as a gamemode, at round start yet, due to other unrelated issues and more recently not enough population, we have noticed no issue which would justify not merging this.

About the issue where people where just being transformed in soul shards: After talking with author he explained that ghosted or disconnected people, or people with mindshield are not supposed to be convertible while they are in such status. After this explanation I am yet to see behavior which I think is actually a bug.

I wanted to make more contributions to bloodcult but haven't been able to but I can help with getting this merged before it gets stale.

Another thing which is confusing about the comments here is it's not clear if they're just comments about bloodcult content in general or about a server running this early merged.

So TLDR; I recommend merging this as it does not break the game, has important bugfixes and is prone to getting stale.

**Comment by rbertoche** [comment]
There. I will now test the merge did not introduce any new weirdness.

**Comment by rbertoche** [comment]
If you prefer I can PR it directly here instead.

**Comment by rbertoche** [comment]
Tests are ok. Converting still works after the merge, which was expected.

Throwing cultist items work as they should, I.E: Cultists can throw after new throwing changes, while non-Cultists get stunned when they try.

As a side note, converting dionas is not possible which seems intended. I'm not sure what's the species white/blacklist.

File: Content.Server/WhiteDream/BloodCult/ConstructActionsSystem.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+using Robust.Shared.Map.Components;
+using PhaseShiftedComponent = Content.Shared.WhiteDream.BloodCult.Constructs.PhaseShift.PhaseShiftedComponent;
+
+namespace Content.Server.WhiteDream.BloodCult;
+
+public sealed class ConstructActionsSystem : EntitySystem
+{
+    [Dependency] private readonly ITileDefinitionManager _tileDef = default!;
+
+    [Dependency] private readonly AudioSystem _audio = default!;
```
Comment (sleepyyapril) [inline Content.Server/WhiteDream/BloodCult/ConstructActionsSystem.cs:16]: you can remove the whitespace yeah?

Comment (Remuchi) [reply Content.Server/WhiteDream/BloodCult/ConstructActionsSystem.cs:16]: My religion tells my otherwise. The whitespace here is to logically separate `Managers` from `Systems`

Comment (sleepyyapril) [reply Content.Server/WhiteDream/BloodCult/ConstructActionsSystem.cs:16]: I respect your religion then

File: Content.Server/WhiteDream/BloodCult/BloodRites/BloodRitesSystem.cs [hunk 2]
```diff
@@ -234,4 +238,36 @@ Entity<BloodstreamComponent> target
...
+
+        var lookup = _lookup.GetEntitiesInRange<PuddleComponent>(
+            coords,
+            rites.Comp.PuddleConsumeRadius,
+            LookupFlags.Uncontained);
+
+        foreach (var puddle in lookup)
+        {
+            if (!TryComp(puddle, out SolutionContainerManagerComponent? solutionContainer))
+                return;
```
Comment (VMSolidus) [inline Content.Server/WhiteDream/BloodCult/BloodRites/BloodRitesSystem.cs:254]:
```suggestion
                continue;
```
Otherwise it will exit early the first time it encounters a puddle without a solution, which while it shouldn't happen, would mean it immediately stops checking all the rest of the puddles.

File: Content.Server/WhiteDream/BloodCult/Runes/CultRuneBaseSystem.cs [hunk 3]
```diff
@@ -85,18 +117,26 @@ private void OnDrawRune(Entity<BloodCultistComponent> ent, ref DrawRuneDoAfter a
         DealDamage(args.User, runeSelector.DrawDamage);
 
         _audio.PlayPvs(args.EndDrawingSound, args.User, AudioParams.Default.WithMaxDistance(2f));
-        var rune = SpawnRune(args.User, runeSelector.Prototype);
+        var runeEnt = SpawnRune(args.User, runeSelector.Prototype);
+        if (TryComp(runeEnt, out CultRuneBaseComponent? rune) && rune.TriggerRendingMarkers)
+        {
+           if (!_cultRule.TryConsumeNearestMarker(ent))
+               return;
+        }
```
Comment (VMSolidus) [inline Content.Server/WhiteDream/BloodCult/Runes/CultRuneBaseSystem.cs:125]:
```suggestion
        if (TryComp(runeEnt, out CultRuneBaseComponent? rune) 
            && rune.TriggerRendingMarkers
            && !_cultRule.TryConsumeNearestMarker(ent))
            return;
```

**Review by VMSolidus — CHANGES_REQUESTED** [review]

**Comment by VMSolidus** [comment]
image

Oh wait that's a weird but legitimate test fail. Huhh?

**Comment by Erisfiregamer1** [comment]
Heisentest?

**Comment by Remuchi** [comment]
I hate obscure test fails

**Comment by whateverusername0** [comment]
image
all maints worldwide preparing to copy cult into their own base

**Comment by Remuchi** [comment]
I swear to god each and every time I look into build fail it gives me new errors. Can't fix until tomorrow sadly so if anyone from maints can fix it for me I'll appreciate it 
image

**Comment by Remuchi** [comment]
> Replace ComponentInit or ComponentStartup with MapInitEvent.

I simply need to specify these thing properly in the Yaml but I have no time until tomorrow

**Comment by VMSolidus** [comment]
image

Almost, just 2 more prototypes. :)
I'll go fix them!

**Comment by Tryded** [comment]
Are null rods added?

**Comment by BlueHNT** [comment]
> Are null rods added?

1. You're month late.
and
2. No.
